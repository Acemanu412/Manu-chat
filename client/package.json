{
  "name": "client",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "jest": {
    "setupFilesAfterEnv": [
      "./setupTests.js"
    ]
  },
  "scripts": {
    "build": "webpack --mode production",
    "build:dev": "webpack --mode development",
    "start": "webpack-dev-server --open --mode development",
    "start:live": "webpack-dev-server --open --mode development --liveReload",
    "test": "jest"
  },
  "keywords": [],
  "author": "Manuel N. Aceituno",
  "license": "ISC",
  "devDependencies": {
    "@babel/core": "^7.11.1",
    "@babel/preset-env": "^7.11.0",
    "@babel/preset-react": "^7.10.4",
    "babel-jest": "^26.3.0",
    "babel-loader": "^8.1.0",
    "cross-fetch": "^3.0.5",
    "css-loader": "^4.2.1",
    "enzyme": "^3.11.0",
    "enzyme-adapter-react-16": "^1.15.3",
    "html-webpack-plugin": "^4.3.0",
    "jest": "^26.4.1",
    "jest-environment-enzyme": "^7.1.2",
    "regenerator-runtime": "^0.13.7",
    "style-loader": "^1.2.1",
    "webpack": "^5.0.0-beta.22",
    "webpack-cli": "^3.3.12",
    "webpack-dev-server": "^3.11.0"
  },
  "dependencies": {
    "@apollo/client": "^3.1.3",
    "graphql": "^15.3.0",
    "react": "^16.13.1",
    "react-dom": "^16.13.1",
    "shards-react": "^1.0.3",
    "subscriptions-transport-ws": "^0.9.18"
  }
}
