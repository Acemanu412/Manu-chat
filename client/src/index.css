body {
    font-family: Arial, Helvetica, sans-serif;
  }

.container{
  min-height: 90vh;
  display:flex;
  flex-flow: column;
}

.input{
  margin-top: auto;
  margin-left: auto;
  margin-right: auto;
  padding: 0 15px 0 15px;
  width: 100%;
}

.message {
  display: flex;
  border-radius: 1em;
  margin: 1em;
}

.messageForm{
  flex-grow:10;
}

.message.mine{
  justify-content: flex-end;
}

.message.notMine{
  justify-content: flex-start;
}

.messageContent{
  max-width: 50%;
  padding: 1em;
  border-radius: 1em;
}
.messageContent.mine {
  background: #6D454C;
  color: #D7FFAB
}

.messageContent.notMine {
  background: #96ADC8;
  color: #0D1B1E;
}

.userLetters {
  height: 50;
  width: 50;
  border: solid 3px #8B635C;
  border-radius: 25px;
  text-align: center;
  font-size: 18;
  padding-top:7px;
  margin-left: 1em;
  margin-right: 1em;
}

.userForm{
  flex-grow: 1;
}