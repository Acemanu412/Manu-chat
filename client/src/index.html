<html>
    <head>
        <link rel="shortcut icon" href="#">
    </head>
    <body>
      <div id="app"></div>
    </body>
</html>