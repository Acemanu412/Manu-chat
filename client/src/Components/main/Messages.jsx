import React from 'react';
import { Alert } from 'shards-react';
import { GET_MESSAGES } from '../../Queries/messages';
import {  useSubscription } from '@apollo/client';

const Messages = ({ user }) => {
    const { loading, error, data } = useSubscription(GET_MESSAGES)

    if (loading) return <Alert theme='info'>Loading...</Alert>;
    if (error) return <Alert theme='danger'>Error!</Alert>;

    return (data.messages.map(({id, user: messageUser, content}) => (
      <div key={id} className={`message ${user === messageUser ? 'mine' : 'notMine'}`}> 
        {user !== messageUser && (<div className='userLetters'>{messageUser.slice(0,2).toUpperCase()}</div>)}
        <div className={`messageContent ${user === messageUser ? 'mine' : 'notMine'}`}>{content}</div>
      </div>
    )))
};

export default Messages;