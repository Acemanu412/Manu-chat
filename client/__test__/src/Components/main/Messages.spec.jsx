import React from 'react';
import {  mount } from 'enzyme';
import { MockedProvider } from '@apollo/client/testing';
import { act } from 'react-dom/test-utils';

import Messages from '../../../../src/Components/main/Messages';
import mocks from '../../../../__mocks__/messages-mocks';

describe('Messages component', () => {
    const user = 'Manu';
    const addTypename = false;
    const wrapper = mount(<MockedProvider mocks={mocks} addTypename={addTypename}>
      <Messages user={user}/>
  </MockedProvider>);
  
  it('should render', () => {
    expect(wrapper.exists()).toBeTruthy();
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render as many messages as messages received from the server', async () => {
    const numberOfMessages = mocks[0].result.data.messages.length;

    await act( async() => await new Promise(resolve => setTimeout(resolve, 0)));

    wrapper.update();

    expect(wrapper.find('.message').length).toBe(numberOfMessages);
  }); 

  it('should render two divs with the class .mine when the user matches the one in the Mock message', () => {
    expect(wrapper.find('.mine').length).toBe(2);
    expect(wrapper.find('.notMine').length).toBe(0);
  })

  const wrapperChangedUser = mount(<MockedProvider mocks={mocks} addTypename={addTypename}>
                                    <Messages user={'Nico'}/>
                                  </MockedProvider>);

  it('should render two divs with the class .notMine when the user does not match the one in the Mock message', async () => {
    await act( async() => await new Promise(resolve => setTimeout(resolve, 0)));
    wrapperChangedUser.update();

    expect(wrapperChangedUser.find('.notMine').length).toBe(2);
    expect(wrapperChangedUser.find('.mine').length).toBe(0);
  })

  it('should render a div with a text equal to the first two letters of the messageUser in uppercase', async () => {
    await act( async() => await new Promise(resolve => setTimeout(resolve, 0)));
    wrapperChangedUser.update();

    expect(wrapperChangedUser.find('.userLetters').text()).toBe(user.slice(0,2).toUpperCase());
  })
});