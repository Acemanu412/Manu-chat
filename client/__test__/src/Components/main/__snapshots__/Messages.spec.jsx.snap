// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Messages component should render 1`] = `"<div class=\\"alert alert-info fade\\" role=\\"alert\\">Loading...</div>"`;
